### PR TITLE
ii-11492 Agent part for enhance performance of federated learning

### PR DESCRIPTION
**Pull request by Zinan-Zhang**

### CODE REVIEW

File: event_push/event_push.py [hunk 1]
```diff
@@ -544,34 +712,76 @@ def worker_process(args):
...
 
-    if not c_config['testing']:
-        # check project name first
-        check_success = check_project_exist(logger, main_vars)
-        if not check_success:
-            return
+    # TODO: get debug info
+    get_debug_info(logger, c_config, edge_vars, main_vars, if_config_vars)
+
+    if c_config['debug-project']:
```
Comment (dearmore) [inline event_push/event_push.py:718]: Need to add the list-projects condition here. Currently the list-projects will continue to process the data after the debug call is made.

Comment (Zinan-Zhang) [reply event_push/event_push.py]: > Need to add the list-projects condition here. Currently the list-projects will continue to process the data after the debug call is made.

Added

**Comment by dearmore** [comment]
regex and statistics packages are still in a .tar.gz format instead of a wheel format which will require building. We should update to use the wheel packages for offline install without building.

**Comment by dearmore** [comment]
install_cron.sh is also still using the old flow without the monitor flag

**Review by dearmore — APPROVED** [review]

**Pull request merged**